src/Driver/PDO/SQLSrv/Connection.php

```diff
@@ -42,15 +43,29 @@ public function exec(string $sql): int
         return $this->connection->exec($sql);
     }
 
-    public function lastInsertId(?string $name = null): string
+    /**
+     * {@inheritDoc}
+     */
+    public function lastInsertId()
     {
-        if ($name === null) {
-            return $this->connection->lastInsertId($name);
-        }
+        return $this->connection->lastInsertId();
+    }
 
-        return $this->prepare('SELECT CONVERT(VARCHAR(MAX), current_value) FROM sys.sequences WHERE name = ?')
+    /**
+     * {@inheritDoc}
+     */
+    public function getSequenceNumber(string $name)
+    {
+        $sequenceNumber =
+            $this->prepare('SELECT CONVERT(VARCHAR(MAX), current_value) FROM sys.sequences WHERE name = ?')
```

Comment:
I'm curious why this `CONVERT(VARCHAR(MAX), ...)` is needed and is it related to the `string|int` type.

Reply:
Looks like this was to respect the `: string` return type, but then I'm not sure why the cast was not done in PHP? I'll try to rework this if you confirm that we're going the `int|string` route.

Reply:
> I'll try to rework this if you confirm that we're going the int|string route.

Let's do.

Reply:
Done in befc0dd4937a53fb8abfa16e9c5403a30ee0d5c5!